pkg/installer/repository.go

```diff
@@ -554,18 +566,40 @@ PACKAGE:
 
 }
 
-func (re Repositories) Search(s string) []PackageMatch {
+func (re Repositories) SearchPackages(p string, o LuetSearchOpts) []PackageMatch {
 	sort.Sort(re)
-	var term = regexp.MustCompile(s)
 	var matches []PackageMatch
+	var err error
 
 	for _, r := range re {
-		for _, pack := range r.GetTree().GetDatabase().World() {
-			if term.MatchString(pack.GetName()) {
+		var repoMatches []pkg.Package
+		if o.Mode == SRegexPkg {
```

Comment:
minor nitpick: this could be easily a `switch`, but no need to fix it here, we can do it in a later batch of changes

Reply:
Done with 40830ecf42bade32001bda396651a9ce27fa5c7f